daml-lf/parser/src/main/scala/com/digitalasset/daml/lf/testing/parser/ExprParser.scala

```diff
@@ -384,6 +385,22 @@ private[parser] class ExprParser[P](parserParameters: ParserParameters[P]) {
       updateGetTime |
       updateEmbedExpr
 
+  private lazy val int: Parser[Int] =
+    acceptMatch("Int", { case Number(l) => l.toInt })
+
+  private lazy val eLoc: Parser[Expr] =
+    `loc` ~>! (`(` ~> dottedName) ~ (`,` ~> id) ~ (`,` ~> int) ~ (`,` ~> int) ~ (`,` ~> int) ~ (`,` ~> int) ~ (`)` ~> expr0) ^^ {
```

Comment:
Could this be  `expr` instead of `expr0`?

Reply:
Are you asking about replacing the `expr0` in this line with `expr` or making this rule part of `expr` instead of `expr0` further up?

The answer to the former is "no". I want the `loc(...)` construct to look like a function and arguments are of the `expr0` form.

The answer to the second would be "good catch". In fact, `eLoc` should be part of `expr` since it looks like an application. Otherwise expressions like `loc(...) f g` would be parsed as `(loc(...) f) g`, which seems somewhat counterintuitive. When `eLoc` is part of `expr`, this expression will be rejected.

I moved `eLoc` to `expr` and added a test case demonstrating that the expression above gets rejected.